src/target/source/interface_rust.cc

```diff
@@ -0,0 +1,381 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
```

Comment:
Would be great to rename to interface_embedded_rust.cc, or place it in a separate namespace.

Both approach would enable deferred changes in followups if necessary while not disrupting existing understanding of the code base in the meantime.

Reply:
if desirable, you are also more than welcome to bring the set of changes to a branch, in which case feel free to continue and complete the work along with followups, and we can take a look again when you feel they are ready.